### PR TITLE
add gas calculation for the basic operations

### PR DESCRIPTION
**Pull request by cyjseagull**

### CODE REVIEW

File: libprecompiled/PrecompiledGas.cpp [hunk 1]
```diff
@@ -24,24 +24,33 @@
...
+    if (_newMemSize > m_memUsed)
+    {
+        m_memUsed = _newMemSize;
+    }
+}
+
+u256 PrecompiledGas::calTotalGas()
 {
-    return (calComputationGas() + calMemGas(_maxMemSize));
+    return (calComputationGas() + calMemGas(m_memUsed));
```
Comment (wheatli) [inline libprecompiled/PrecompiledGas.cpp]: unnecessary to pass member variable to a member function

File: libprecompiled/Precompiled.cpp [hunk 2]
```diff
@@ -69,3 +69,15 @@ bool Precompiled::checkAuthority(std::shared_ptr<dev::blockverifier::ExecutiveCo
     }
     return true;
 }
+
+uint64_t Precompiled::getEntriesCapacity(
+    std::shared_ptr<dev::storage::Entries const> _entries) const
+{
+    int64_t totalCapacity = 0;
+    int64_t entriesSize = _entries->size();
+    for (int64_t i = 0; i < (entriesSize - 1); i++)
```
Comment (wheatli) [inline libprecompiled/Precompiled.cpp]: why 'i < (entriesSize - 1)'

File: libprecompiled/PrecompiledGas.cpp (hunk 1 above)
Comment (cyjseagull) [reply libprecompiled/PrecompiledGas.cpp]: Fixed.

File: libprecompiled/Precompiled.cpp (hunk 2 above)
Comment (cyjseagull) [reply libprecompiled/Precompiled.cpp]: Fixed

**Pull request merged**